routes/actionsRouter.js

```diff
@@ -0,0 +1,70 @@
+const express = require("express");
+const route = express.Router();
+
+const actionModel = require("../data/helpers/actionModel");
+//middleware
+const checkAction = require("../common/checkAction");
+
+function generalErr(err, res) {
+  res
+    .status(500)
+    .json({ message: "There is something wrong is the database." });
+}
+
+route.get("/", async (req, res) => {
+  try {
+    const actionList = await actionModel.get();
+    res.json(actionList);
+  } catch (err) {
+    generalErr(err, res);
+  }
+});
+
+route.post("/add", checkAction, async (req, res) => {
+  const action = req.body;
+  try {
+    const result = await actionModel.insert(action);
+    if (result) {
+      res.status(201).json({ message: "Action has been created!" });
+    }
+  } catch (err) {
+    generalErr(err, res);
+  }
+});
+
+route.put("/:id", checkAction, async (req, res) => {
+  const { id } = req.params;
+  const change = req.body;
+  try {
+    const action = await actionModel.get(id);
+    if (action) {
+      const result = await actionModel.update(id, change);
+      res.json({
+        message: `Action has been updated and has an id of ${result.id}`
+      });
+    } else {
+      res.status(404).json({ message: "Action not found!" });
+    }
+  } catch (err) {
+    generalErr(err, res);
+  }
+});
+
+route.delete("/:id", async (req, res) => {
+  const { id } = req.params;
+  try {
```

Comment:
Love the async/await syntax, your in the minority as far as who is using that on this project but it's the more modern way to write promises and there's a reason the implemented it. (I also see a lot of my courses using it)